### PR TITLE
 feat(discover): Add conditions UI and basic validation

### PR DESCRIPTION
**Pull request by lynnagara**
Closes APP-391

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    if (isValidCondition(external, this.props.columns)) {
+      this.props.onChange(external);
+      return;
+    }
+
+    if (new Set(this.props.columns.map(({name}) => name)).has(external[0])) {
+      this.setState({selectedColumn: external[0]}, this.focus);
+    }
+
+    if (new Set(CONDITION_OPERATORS).has(external[1])) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: why not just `Array.includes`?

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import SelectControl from 'app/components/forms/selectControl';
+
+import {getInternal, getExternal, isValidCondition} from './utils';
+import {CONDITION_OPERATORS} from '../data';
+
+export default class Condition extends React.Component {
+  static propTypes = {
+    value: PropTypes.array,
+    onChange: PropTypes.func,
+    columns: PropTypes.array,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: Do columns have a type we can define with `PropTypes.shape` here?

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    return isValidCondition(getExternal(label, this.props.columns), this.props.columns);
+  };
+
+  inputRenderer = props => {
+    let val = `${this.state.selectedColumn || ''} ${this.state.selectedOperator ||
+      ''}`.trim();
+
+    return (
+      <input
+        id="custom-input"
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: Is this used for anything?

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        type="text"
+        {...props}
+        value={props.value || val}
+        style={{width: '100%', border: 0}}
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: can we change this to be a styled component?

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/index.jsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        {!value.length && 'None, showing all events'}
+        {value.map((condition, idx) => (
+          <Flex key={idx}>
+            <Condition
+              value={condition}
+              onChange={val => this.handleChange(val, idx)}
+              columns={columns}
+            />
+            <Box ml={1}>
+              <a
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/conditions/index.jsx]: We should use the svg version + styled components

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/index.jsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import {Flex, Box} from 'grid-emotion';
+
+import Link from 'app/components/link';
+import {t} from 'app/locale';
+
+import Condition from './condition';
+
+export default class Conditions extends React.Component {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/conditions/index.jsx]: It'd be nice to have tests for add row, remove row, and on change

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx [hunk 7]
```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Box} from 'grid-emotion';
+
+import SelectControl from 'app/components/forms/selectControl';
+
+import {getInternal, getExternal, isValidCondition} from './utils';
+import {CONDITION_OPERATORS} from '../data';
+
+export default class Condition extends React.Component {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: It'd be good to have tests for on change, filter options, and on input change

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: force of habit, and using sets elsewhere here

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx (hunk 2 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: yep

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx (hunk 3 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: no, removing

File: src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx (hunk 4 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx]: Doesn't play nice with react select

**Review by billyvg — APPROVED** [review]

**Pull request merged**